flocker/node/test/test_deploy.py

```diff
@@ -2324,27 +2349,18 @@ def test_dataset_id_populated(self):
                 applications=frozenset([APPLICATION_WITH_VOLUME_SIZE])
             )
         ]))
-        api = P2PNodeDeployer(u"node", create_volume_service(self),
-                              docker_client=FakeDockerClient(),
+
+        api = P2PNodeDeployer(u"node", volume_service,
+                              docker_client=docker,
                               network=make_memory_network())
         result = api.calculate_necessary_state_changes(
             self.successResultOf(api.discover_local_state()), desired, actual)
 
         # Desired configuration was changed to set the correct dataset ID,
         # which is why a resize is happening:
```

Comment:
I don't understand this. I know it's not related to this branch, but could you add some further information about why updating dataset_id causes a ResizeDataset step?